tests/test_pay.py

```diff
@@ -2074,62 +2089,61 @@ def test_setchannelfee_routing(node_factory, bitcoind):
     scid = l2.get_channel_scid(l3)
 
     # TEST CUSTOM VALUES
-    l2.rpc.setchannelfee(scid, 1337, 137)
+    l2.rpc.setchannel(scid, 1337, 137, 4000000, enforcedelay=0)
 
     # wait for l1 to see updated channel via gossip
     wait_for(lambda: [c['base_fee_millisatoshi'] for c in l1.rpc.listchannels(scid)['channels']] == [1337, DEF_BASE])
     wait_for(lambda: [c['fee_per_millionth'] for c in l1.rpc.listchannels(scid)['channels']] == [137, DEF_PPM])
+    wait_for(lambda: [c['htlc_maximum_msat'] for c in l1.rpc.listchannels(scid)['channels']] == [4000000, MAX_HTLC])
 
     # test fees are applied to HTLC forwards
     #
     # BOLT #7:
     # If l1 were to send 4,999,999 millisatoshi to l3 via l2, it needs to
     # pay l2 the fee it specified in the l2->l3 `channel_update`, calculated as
     # per [HTLC Fees](#htlc_fees):  base + amt * pm / 10**6
-    #
-    # 1337 + 4999999 * 137 / 1000000 = 2021.999 (2021)
-    route = l1.rpc.getroute(l3.info['id'], 4999999, 1)["route"]
-    assert len(route) == 2
-    assert route[0]['msatoshi'] == 5002020
-    assert route[1]['msatoshi'] == 4999999
 
+    # FIXME!
+#    # Should refuse to route this!
+#    with pytest.raises(RpcError, match=r'Could not find a route'):
+#        l1.rpc.getroute(l3.info['id'], 4000001, 1, fuzzpercent=0)["route"]
+
+    # 1337 + 4000000 * 137 / 1000000 = 1885
+    route_ok = l1.rpc.getroute(l3.info['id'], 4000000, 1)["route"]
```

Comment:
Just so we don't forget it, if you go for `4000000` in this test also update the comment line  2102:
`If l1 were to send 4,999,999 millisatoshi to l3 via l2`

Reply:
Maybe also go for 3,999,999 to provoke potential rounding errors